### DISCUSSION TITLE
NextJS for backend

### DISCUSSION THREAD

**Discussion by Louai99k** [vercel/next.js#31810] [Help] [answered]
Is Next JS Backend side (API routes and middlewares) is enough for full-stack website or should I use custom Nodejs server and if it so what is the benefits I will get from the custom server (What it will give me that api routes and middlewares can't?)

**Comment by jamesmosier** [comment, marked as answer]
API routes and middleware from Next.js should be enough for your project. There are many more disadvantages to using a custom Node server than benefits. I personally see no reason to use a custom Node server anymore, especially with the release of [middleware](https://nextjs.org/docs/middleware).

**Reply by Louai99k** [reply]
Broo thanks so much some people told me that things like event triggering or sending mail cannot be done by Nextjs backend is this true??

**Reply by jamesmosier** [reply]
> event triggering 

Can you elaborate on this a bit?


> sending mail 

I'm not sure of your setup for this, but I have not had any issues doing this. I typically use a third-party service though, but something like nodemailer should work just fine.

**Reply by Louai99k** [reply]
So you're saying that you use Nextjs for full stack projects and it works just fine.
This answer is enough for me thanks a lot.
